torch/utils/module_tracker.py

```diff
@@ -85,49 +135,56 @@ def _get_mod_name(self, mod):
             self._seen_modules.add(mod)
         return mod_name
 
-    def _get_append_fn(self, name, is_bw):
+    def _get_append_fn(self, mod, name, is_bw):
```

Comment:
I need a mod reference in my hook, so either I get the module reference directly or have a reverse weak value dictionary from fqn->module? What do you think?

Reply:
you don't need a full reverse table, you can have this function take a weak_mod and materialize it when calling the user hook.
But you need to know what to do if the module is gone by that time.

Reply:
@albanD But if the module instance gets deleted I will lose all of the snapshots in memory tracker because we store them by weak refs to module's in a dictionary

Reply:
Right now, added the weak refs as args to the append and pop functions. They materialize the ref before passing it to the user hook. If the user hook receives None, it's backward snapshot is not captured and a warning is raised that the module is not alive, so skipping snapshot.